test/extensions/clusters/aggregate/cluster_test.cc

```diff
@@ -169,6 +190,351 @@ class AggregateClusterTest : public Event::TestUsingSimulatedTime, public testin
 )EOF";
 }; // namespace Aggregate
 
+TEST_F(AggregateClusterTest, CircuitBreakerDefaultsTest) {
+  initialize(default_yaml_config_);
+
+  Upstream::ResourceManager& resource_manager =
+      cluster_->info()->resourceManager(Upstream::ResourcePriority::Default);
+
+  // the default circuit breaker values are:
+  // max_connections : 1024
+  // max_pending_requests : 1024
+  // max_requests : 1024
+  // max_retries : 3
+
+  EXPECT_EQ(1024U, resource_manager.connections().max());
+  for (int i = 0; i < 1024; ++i) {
+    resource_manager.connections().inc();
+  }
+  EXPECT_EQ(1024U, resource_manager.connections().count());
+  EXPECT_FALSE(resource_manager.connections().canCreate());
+
+  EXPECT_EQ(1024U, resource_manager.pendingRequests().max());
+  for (int i = 0; i < 1024; ++i) {
+    resource_manager.pendingRequests().inc();
+  }
+  EXPECT_EQ(1024U, resource_manager.pendingRequests().count());
+  EXPECT_FALSE(resource_manager.pendingRequests().canCreate());
+
+  EXPECT_EQ(1024U, resource_manager.requests().max());
+  for (int i = 0; i < 1024; ++i) {
+    resource_manager.requests().inc();
+  }
+  EXPECT_EQ(1024U, resource_manager.requests().count());
+  EXPECT_FALSE(resource_manager.requests().canCreate());
+
+  EXPECT_EQ(3U, resource_manager.retries().max());
+  for (int i = 0; i < 3; ++i) {
+    resource_manager.retries().inc();
+  }
+  EXPECT_EQ(3U, resource_manager.retries().count());
+  EXPECT_FALSE(resource_manager.retries().canCreate());
+}
```

Comment:
There's a good chance that the envoy reviewer will ask for this test to be removed.

The reason is that if they ever want to change the defaults of the circuit breaker, they'll want to only have tests in the circuit breaker affected. Whereas adding this test would mean if they change the circuit breaker, this test will fail.

You're fine to leave it in, just a heads-up that they may ask to remove this. :heart:

Reply:
Yes, valid point 👍
See https://github.com/bazmurphy/envoy/pull/14#discussion_r1933466438